proto/regen/ecocredit/v1/types.proto

```diff
@@ -34,3 +34,13 @@ message Params {
   // credit_types is a list of definitions for credit types
   repeated CreditType credit_types = 5;
 }
+
+// OriginTx is a reference to an external transaction or an operation
+// related to an action on Regen Ledger.
+message OriginTx {
+  // type of the transaction originating the mint process. Eg: Polygon,
+  // Ethereum, Verra,
+  string typ = 1;
+  // the id of a transaction based on a type (tx id, serial number)
+  string id = 2;
```

Comment:
Normally I would prefer to  not storing everything. We have events to keep that information. But events are not guaranteed to be permanent. 
So ideally we don't store everything here and we sort out the events in SDK.

What do you think?